07-arrays-dictionaries-sets/projects/challenge/arrays-dictionaries-sets-challenges.playground/Contents.swift

```diff
@@ -126,10 +126,15 @@ func reversed(_ array: [Int]) -> [Int] {
 */
 
 func middle(_ array: [Int]) -> Int? {
-  guard !array.isEmpty else {
-    return nil
-  }
-  return array[array.count/2]
+    guard !array.isEmpty else {
+        return nil
+    }
+    if array.count % 2 == 0{
```

Comment:
A little late to the party here. You can actually do (array.count-1)/2 for all cases, I believe. That's because integer division is always truncating. We can address it in the next edition.  Thanks!

Reply:
Thanks again.  I am going to close this PR (if it will let me) and it will be fixed in the next edition.  Sorry for not looking here earlier.